woocommerce/wc-admin
# Refactor/consolidate Datasource poller

This is a follow up from another PR and in particular this comment: https://github.com/woocommerce/woocommerce-admin/pull/7553#discussion_r698730821

I moved all the overlapping DataSourcePoller logic into one class, the easiest way was to create a non static class. I also moved the transient logic into the DataSourcePoller which originally was handled outside of it.

### Detailed test instructions:

- On a new site with this branch loaded
- Start the onboarding wizard (have your console open) and make sure the free features are listed on the business details step
- Check the `free-extensions` request and make sure the `image_url` of some of the plugins (under `grow` or `reach`) are coming from `woocommerce.com`.
- Finish the onboarding wizard and go to the Marketing task, this should load correctly and the images displayed should come from `woocommerce.com` (note: if they are coming from your local something went amiss in the data source polling)
- Go to the **Set up payments** task and check for the image urls as well
- Go to **WooCommerce > Settings > Payments** make sure the recommended payment card shows up underneath the table.
- Run the `wc_admin_daily` job (you can use [WC test helper](https://github.com/woocommerce/woocommerce-admin-test-helper/) or [WP Crontrol](https://en-ca.wordpress.org/plugins/wp-crontrol/) ) and go to **WooCommerce > Home** check the notes and make sure some of the woocommerce.com notes are displayed from https://woocommerce.com/wp-json/wccom/inbox-notifications/1.0/notifications.json you can also check the `wp_wc_admin_notes` table and see if it contains notes with the source `woocommerce.com`.
